### PR TITLE
Allow access_type parameter to be overriden

### PR DESCRIPTION
**Pull request by jonparrott**

### CODE REVIEW

File: google_auth_oauthlib/flow.py [hunk 1]
```diff
@@ -107,6 +112,7 @@ def __init__(self, oauth2session, client_type, client_config):
         """Mapping[str, Any]: The OAuth 2.0 client configuration."""
         self.oauth2session = oauth2session
         """requests_oauthlib.OAuth2Session: The OAuth 2.0 session."""
+        self.redirect_uri = redirect_uri
```
Comment (dhermes) [inline google_auth_oauthlib/flow.py]: Do you want a docstring for this?

File: google_auth_oauthlib/flow.py [hunk 2]
```diff
@@ -200,9 +206,9 @@ def authorization_url(self, **kwargs):
                 :class:`Flow` instance to obtain the token, you will need to
                 specify the ``state`` when constructing the :class:`Flow`.
         """
+        kwargs.setdefault('access_type', 'offline')
```
Comment (dhermes) [inline google_auth_oauthlib/flow.py]: Probably worth updating the docstring to explain that `access_type=offline` will be added if not passed in?

File: google_auth_oauthlib/flow.py [hunk 3]
```diff
@@ -229,10 +235,9 @@ def fetch_token(self, **kwargs):
                 :meth:`credentials` to obtain a
                 :class:`~google.auth.credentials.Credentials` instance.
         """
+        kwargs.setdefault('client_secret', self.client_config['client_secret'])
```
Comment (dhermes) [inline google_auth_oauthlib/flow.py]: `KeyError` ouch?

File: google_auth_oauthlib/flow.py (hunk 1 above)
Comment (jonparrott) [reply google_auth_oauthlib/flow.py]: There's already one for it below (it's a property)

File: google_auth_oauthlib/flow.py (hunk 2 above)
Comment (jonparrott) [reply google_auth_oauthlib/flow.py]: It's there.

File: google_auth_oauthlib/flow.py (hunk 3 above)
Comment (jonparrott) [reply google_auth_oauthlib/flow.py]: It's okay if that happens (for now at least)

Comment (dhermes) [reply google_auth_oauthlib/flow.py]: :+1:

File: google_auth_oauthlib/flow.py (hunk 2 above)
Comment (dhermes) [reply google_auth_oauthlib/flow.py]: D'oh

**Pull request merged**